lib/api/documentExtractor.js

```diff
@@ -154,7 +157,27 @@ const extractors = ([
         return request;
       }
     },
-    targets: ['mCreate', 'mCreateOrReplace', 'mReplace', 'mUpdate']
+    targets: ['mCreate', 'mCreateOrReplace', 'mReplace', 'mUpdate', 'updateByQuery']
+  },
+  {
+    methods: {
+      extractFromRequest: null, // not eligible for search queries
+      extractFromResult: request => {
+        return request.result.hits || request.result.documents;
+      },
+      insertInRequest: null, // not eligible for search queries
+      insertInResult: (documents, request) => {
+        if (request.input.action === 'search') {
+          request.result.hits = documents;
+        }
+        else if (request.input.action === 'deleteByQuery') {
+          request.result.documents = documents;
+        }
+
+        return request;
+      },
+    },
+    targets: ['search', 'deleteByQuery'],
```

Comment:
I thought `updateByQuery` was not eligible either?

Reply:
`updateByQuery` is handled by the same extractor than `mCreate`, `mReplace`, etc. because they share the same format.

The blacklist is done in the `documentEventAliases.js` file: https://github.com/kuzzleio/kuzzle/pull/1698/files#diff-cda9b5cb3d677fb4f72bd365d3318910

So the `extractFromRequest` and `insertInRequest` extractor methods for `updateByQuery` cannot be invoked.